lib/config.ml

```diff
@@ -255,6 +247,20 @@ let keep_v v = if v then Ok true else read (fun t -> t.keep_v) false
 
 let auto_open v = if not v then Ok false else read (fun t -> t.auto_open) true
 
+let opam_repo_fork ?pkgs ~remote ~local () =
```

Comment:
I think it would be nice to find a way to distinguish the values that are only the CLI options and the ones that are the unified values (CLI options / file config), because it's a bit confusing to have `local`/`remote` values in input and in output.

Reply:
Yeah I agree, do you have something in particular in mind? Would renaming the labels be enough for instance or do you have something more involved in mind?

Reply:
I was thinking of something more involved (to be done in a separate PR), like having a type `'a t` in the cli for all options, same for values returned in config.ml, and then unification (not sure it's the right term) functions would have type like: `string Cli.t -> string Config.t -> string`

Reply:
Yeah that's a good idea! I can give it a quick try. What would you think about this:

Since it only matters for config, I can extract just a `Config.Cli.t` type with a smart constructor, keeping it abstract. I then use this constructor for the CLI args `keep_v`, `no_auto_open`, `local_repo` and `remote_repo`. The only way to consume those then is to pass them to the right functions in `Config`.

That's fairly small so I can include it in this PR and it feels like it would solve our problem here.

Reply:
Ok I'm working on this but the change is slightly more involved than I hoped it would be so let's deal with it in a separate PR.